gallery/tutorials/aspire_introduction.py

```diff
@@ -0,0 +1,675 @@
+#
+# The white noise estimator should log a diagnostic variance value.
+# Internally, it also uses the estimation results to build a
+# ``Filter`` which can be used in more advanced denoising methods.
+
+from aspire.noise import WhiteNoiseEstimator
+
+noise_estimator = WhiteNoiseEstimator(sim)
+noise_estimator.estimate()
+
+
+# %%
+# A Custom ``FunctionFilter``
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# We will now apply some more interesting noise, using a custom
+# function, and then apply a ``whitening`` process to our data.
+#
+# Using ``FunctionFilter`` we can create our own custom functions to
+# apply in a pipeline.  Here we want to apply a custom noise function.
+# We will use a function of two variables for this example.
+
+from aspire.noise import CustomNoiseAdder
+from aspire.operators import FunctionFilter
+
+
+def noise_function(x, y):
+    return 1e-7 * np.exp(-(x * x + y * y) / (2 * 0.3**2))
+
+
+# In python, functions are first class objects.  We take advantage of
+# that to pass this function around as a variable.  The function is
+# evaluated later, internally, during pipeline execution.
+custom_noise = CustomNoiseAdder(noise_filter=FunctionFilter(noise_function))
+
+# Create yet another Simulation source to tinker with.
+sim = Simulation(n=num_imgs, vols=v2, noise_adder=custom_noise)
+sim.images[:10].show()
+
+# %%
+# Noise Whitening
+# ^^^^^^^^^^^^^^^
+# We will now combine a more advanced noise estimation technique with
+# an ``ImageSource`` preprocessing method ``whiten``.
+#
+# First an anisotropic noise estimate is performed.
+
+from aspire.noise import AnisotropicNoiseEstimator
+
+# Estimate noise.
+aiso_noise_estimator = AnisotropicNoiseEstimator(sim)
+
+# %%
+# Applying the ``Simulation.whiten()`` method requires passing a
+# corresponding ``NoiseEstimator`` instance.  Then we can inspect some
+# of the whitened images.  While noise is still present, we can see a
+# dramatic change.
+
+# Whiten based on the estimated noise.
+sim.whiten(aiso_noise_estimator)
+
+# %%
+# What do the whitened images look like?
+sim.images[:10].show()
+
+# %%
+# Common Image Corruptions
+# ------------------------
+# ``Simulation`` provides several configurable types of common CryoEM
+# image corruptions.  Users should be aware that Amplitude and Offset
+# corruption is enabled by default.
+
+# %%
+# Amplitudes
+# ^^^^^^^^^^
+# Simulation automatically generates random amplitude variability.
+# To disable, set to ``amplitudes=1``.
+
+# %%
+# Offsets
+# ^^^^^^^
+# Simulation automatically generates random offsets.
+# To disable, set to ``offsets=0``.
+
+# %%
+# Noise
+# ^^^^^
+# By default, no noise corruption is configured.
+# To enable, see ``NoiseAdder`` components.
+
+# %%
+# CTF
+# ^^^
+# By default, no CTF corruption is configured.
+# To enable, we must configure one or more CTFFilter.
```

Comment:
``CTFFilter``s

Reply:
``CTFFilter`` instances (``s seems broken in gallery,  kind of annoying)